src/use-cases/SaveMetadata.ts

```diff
@@ -21,21 +23,32 @@ interface SaveMetadataResult {
 export default class SaveMetadataUseCase
   implements UseCase<SaveMetadataCommand, SaveMetadataResult> {
   s3Gateway: S3Gateway;
+  esGateway: ElasticsearchGateway;
   createDocumentId: () => string;
 
-  constructor({ s3Gateway, createDocumentId }: SaveMetadataDependencies) {
+  constructor({
+    s3Gateway,
+    createDocumentId,
+    esGateway,
+  }: SaveMetadataDependencies) {
     this.s3Gateway = s3Gateway;
     this.createDocumentId = createDocumentId;
+    this.esGateway = esGateway;
   }
 
   async execute({
     metadata,
   }: SaveMetadataCommand): Promise<SaveMetadataResult> {
+    const documentId = this.createDocumentId();
+
     const created = await this.s3Gateway.create({
-      documentId: this.createDocumentId(),
+      documentId,
       ...metadata,
     });
 
+    const indexMetadata = Object.assign({}, metadata, { documentId });
```

Comment:
```suggestion
    const indexMetadata = {
      ...metadata,
      documentId,
    };
```

should also work, or maybe we could pull the object that gets passed to `.create` out into a variable and use that rather than recreating it?

Reply:
...or maybe `created` already has this value?